pallets/pool-system/src/impls.rs

```diff
@@ -66,13 +66,15 @@ impl<T: Config> TrancheTokenPrice<T::AccountId, T::CurrencyId> for Pallet<T> {
 		pool_id: Self::PoolId,
 		tranche_id: Self::TrancheId,
 	) -> Option<(T::BalanceRatio, Seconds)> {
-		let now = T::Time::now();
 		let mut pool = Pool::<T>::get(pool_id)?;
 
-		// Get cached nav as calculating current nav would be too computationally
-		// expensive
-		let (nav, nav_last_updated) = T::AssetsUnderManagementNAV::nav(pool_id)?;
-		let total_assets = pool.reserve.total.ensure_add(nav).ok()?;
+		let (nav_loans, nav_loans_updates) = T::AssetsUnderManagementNAV::nav(pool_id)?;
+		let (nav_fees, nav_fees_updated) = T::PoolFeesNAV::nav(pool_id)?;
+
+		let nav = Nav::new(nav_loans, nav_fees);
+		let total_assets = nav
+			.total(pool.reserve.total)
+			.unwrap_or(<T as Config>::Balance::zero());
```

Comment:
Super NIT: `unwrap_or_default()`

Reply:
I am not the biggest fan of default in these case. Zero is really explicit and also what we want.